### PR TITLE
feat(web): go back on the onboarding

### PR DESCRIPTION
**Pull request by martabal**
## Changes made in this PR

This PR adds the ability to use the browser history to go through each steps of the onboarding. Also adds a border to show which theme is selected.

## Screenshots

https://github.com/immich-app/immich/assets/74269598/8ebda6fd-a4d8-47f5-b25b-c6b15d4de900

### CODE REVIEW

**Review by alextran1502 — APPROVED** [review]

File: web/src/routes/auth/onboarding/+page.svelte [hunk 1]
```diff
@@ -5,21 +5,38 @@
...
 
-  let onboardingSteps = [OnboardingHello, OnboardingTheme, OnboadingStorageTemplate];
+  interface OnboardingStep {
+    name: string;
+    component: typeof OnboardingHello | typeof OnboardingTheme | typeof OnboadingStorageTemplate;
+  }
 
-  const handleDoneClicked = async () => {
-    index++;
+  let onboardingSteps: OnboardingStep[] = [
```
Comment (jrasm91) [inline web/src/routes/auth/onboarding/+page.svelte:17]: this should probably be `const`, no?

**Review by jrasm91 — COMMENTED** [review]
Would itg be possible to add a back arrow on the onboarding form itself?

File: web/src/routes/auth/onboarding/+page.svelte (hunk 1 above)
Comment (martabal) [reply web/src/routes/auth/onboarding/+page.svelte:17]: Yep

**Comment by martabal** [comment]
> Would itg be possible to add a back arrow on the onboarding form itself?

I think both are good (I almost only use my 4/5 buttons on my mouse)

**Pull request merged**